File: model/src/main/java/net/artux/pda/model/quest/CheckpointModel.java
package net.artux.pda.model.quest;

import java.io.Serializable;

import lombok.Data;

@Data
public class CheckpointModel implements Serializable {
    private String parameter;
    private String title;
    private Integer chapter;
    private Integer stage;

    public boolean isActual(String... params){
        for (String param : params){
            if (param.equals(parameter))
                return true;
        }
        return false;
    }
}

File: model/src/main/java/net/artux/pda/model/quest/CheckpointModel.kt
package net.artux.pda.model.quest

import java.io.Serializable

class CheckpointModel : Serializable {
    var parameter: String? = null
    var title: String? = null
    var chapter: Int? = null
    var stage: Int? = null
    fun isActual(vararg params: String): Boolean {
        for (param in params) {
            if (param == parameter) return true
        }
        return false
    }
}